tests/YearMonthTest.php

```diff
@@ -330,16 +330,38 @@ public function testWithYear()
         $this->assertYearMonthIs(2001, 5, YearMonth::of(2000, 5)->withYear(2001));
     }
 
+    public function testWithYearIsTheSameYear()
+    {
+        $year = 2018;
+        $month = 2;
+        $yearMonth = YearMonth::of($year, $month);
+
+        $this->assertInstanceOf(YearMonth::class, $yearMonth->withYear($year));
+        $this->assertYearMonthIs($year, $month, $yearMonth->withYear($year));
+    }
+
     public function testWithMonth()
     {
         $this->assertYearMonthIs(2000, 12, YearMonth::of(2000, 1)->withMonth(12));
     }
 
+    public function testWithMonthWithSameMonth()
+    {
+        $month = 2;
+
+        $this->assertYearMonthIs(2000, $month, YearMonth::of(2000, $month)->withMonth($month));
+    }
+
     public function testGetFirstDay()
     {
         $this->assertLocalDateIs(2023, 10, 1, YearMonth::of(2023, 10)->getFirstDay());
     }
 
+    public function testGetLastDay()
```

Comment:
Actually there was already a test for getLastDay(), you can remove this method! See below.